### ISSUE TITLE
Batch processing with different values

### ISSUE THREAD

**Issue by bentron2000** [janfri/multi_exiftool#10]
Hi - Just looking at batching changes to a large number of files using this - it doesn't look like a batch of files can receive different values per file.

Would it be difficult to modify `values_args` to take an array of valid values in the same order as the filenames array and have the values zip together as the command is built?

Apologies for not putting in a PR - my ruby is rusty rn

Cheers
b

**Comment by janfri** [comment]
At the moment it's not possible to write different values for different files with MultiExiftool with only one call of the Exiftool command-line application. But I will have a look at it. With the ```-execute``` option it should be possible. Be patient.

You can do it not so performant like the following example:
```ruby
filenames = ['a.jpg', b.jpg', 'c.jpg']
values = [{comment: 'Comment A'}, {comment: 'Comment B'}, {comment: 'Comment C'}]

filenames.zip values do |fn, vals|
  MultiExiftool.write fn, vals
end
```
But this will call the Exiftool command-line application once for each filename.

**Comment by janfri** [comment]
I have just released multi_exiftool 0.13.0. Now you can do batch processing:
```ruby
filenames = ['a.jpg', b.jpg', 'c.jpg']
values = [{comment: 'Comment A'}, {comment: 'Comment B'}, {comment: 'Comment C'}]

batch = MultiExiftool::Batch.new do
  filenames.zip values do |fn, vals|
    write fn, vals
  end
end
if batch.execute
  puts 'ok'
else
  $stderr.puts batch.errors
end
```
or with the funtional api:
```ruby
filenames = ['a.jpg', b.jpg', 'c.jpg']
values = [{comment: 'Comment A'}, {comment: 'Comment B'}, {comment: 'Comment C'}]

errors = MultiExiftool.batch do
  filenames.zip values do |fn, vals|
    write fn, vals
  end
end
if errors.empty?
  puts 'ok'
else
  $stderr.puts batch.errors
end
```
While execution the ExifTool command-line application is only called once thanks to the ```-execute``` option. 😄

**Comment by bentron2000** [comment]
This is fantastic! I will go and try it out now :)

Thank you @janfri :)

**Comment by janfri** [comment]
Let me know if it works for you.

**Comment by janfri** [comment]
So now I have impoved the api for batch processing a litte bit and released multi_exiftool 0.14.0.

Now you can write
```ruby
filenames = ['a.jpg', b.jpg', 'c.jpg']
values = [{comment: 'Comment A'}, {comment: 'Comment B'}, {comment: 'Comment C'}]

batch = MultiExiftool::Batch.new
filenames.zip values do |fn, vals|
  batch.write fn, vals
end
if batch.execute
  puts 'ok'
else
  $stderr.puts batch.errors
end
```
and with the functional api you have two alternatives.

First, the same as above:
```ruby
filenames = ['a.jpg', b.jpg', 'c.jpg']
values = [{comment: 'Comment A'}, {comment: 'Comment B'}, {comment: 'Comment C'}]

errors = MultiExiftool.batch do
  filenames.zip values do |fn, vals|
    write fn, vals
  end
end
if errors.empty?
  puts 'ok'
else
  $stderr.puts batch.errors
end
```
and second with explicit Batch instance as block parameter:
```ruby
filenames = ['a.jpg', b.jpg', 'c.jpg']
values = [{comment: 'Comment A'}, {comment: 'Comment B'}, {comment: 'Comment C'}]

errors = MultiExiftool.batch do |batch|
  filenames.zip values do |fn, vals|
    batch.write fn, vals
  end
end
if errors.empty?
  puts 'ok'
else
  $stderr.puts batch.errors
end
```
The second version is helpful if you want to use instance variables of the outer context.
